lib/events/auditwriter.go

```diff
@@ -375,6 +380,10 @@ func (a *AuditWriter) EmitAuditEvent(ctx context.Context, event apievents.AuditE
 	}
 }
 
+func (a *AuditWriter) EmitAuditEvent(ctx context.Context, event apievents.AuditEvent) error {
+	return errors.New("EmitAuditEvent should not be called from AuditWriter, call EmitSessionRecordingEvent instead")
+}
+
```

Comment:
We shouldn't have to lie about implementing this method. I expect this will explode in runtime. Tests agree: https://github.com/gravitational/teleport/actions/runs/5192709968/jobs/9362251412?pr=27426#step:6:1158

```
{"caller":"srv/sess.go:964","component":"session:node","error":"EmitAuditEvent should not be called from AuditWriter, call EmitSessionRecordingEvent instead","level":"warning","message":"Failed to emit session leave event.","session_id":"4ff533ba-23ea-4264-b2fd-f8ecacfe2482","timestamp":"2023-06-06T19:52:21Z"}
```

Reply:
Yeah I thought at first code using `AuditWriter` only emitted session recording events, but apparently some callers emit audit events too. I changed some interfaces instead to ensure the correct methods are being used.

Reply:
This is an improvement but I wonder if they should actually be separate interfaces..

Reply:
I created separate interfaces for audit and session event emitting as well as some interfaces that allow both, depending on the needs of the caller